### PR TITLE
feat(config): add Gittensory repo focus manifest

### LINKED ISSUE #118: feat(config): add Gittensory repo focus manifest
Parent roadmap: https://github.com/JSONbored/gittensory/issues/82

## Background

If Gittensory attracts miners, they need clear direction toward high-value backend/MCP/GitHub App work rather than website polish or vague cleanup.

## Goal

Add a repo focus manifest that defines wanted work, out-of-scope work, required tests, public-safety boundaries, and preferred modules.

## Current Behavior

Contributor issues can describe focus, but there is no machine-readable project focus policy for Gittensory itself.

## Desired Behavior

Gittensory can feed its own focus manifest into decision packs, local preflight, issue drafting, and maintainer guidance.

## Implementation Requirements

- Define `.gittensory.yml` or API-backed equivalent for this repo.
- Include wanted areas, blocked areas, labels, test expectations, public/private boundaries, and maintainer notes.
- Validate manifest.
- Feed into recommendations.
- Keep private notes private.

## Public/Private Output Boundaries

Self-dogfood reports may include private maintainer/config readiness in authenticated views. Public docs/issues must stay advisory and must not imply guaranteed Gittensor payouts.

## Acceptance Criteria

- Manifest validates.
- Wanted paths influence recommendations.
- Website-only work is discouraged unless safety/release related.
- Malformed manifest warns safely.

## Testing Requirements

- `npm run test:ci` must pass.
- Global coverage must remain at or above 97% for lines, statements, functions, and branches.
- Aim for 98%+ branch coverage locally to avoid CI variance.
- Add tests for every new branch, fallback path, sanitizer rule, and regression.
- Add invariant/property-style tests when behavior depends on sorting, gating, scoring, queue pressure, source-upload safety, public/private boundaries, or upstream drift.
- Public GitHub output must be tested against forbidden language: wallet, hotkey, raw trust score, payout, reward estimate, farming, private reviewability, and public score estimate.
- MCP/local tooling must prove source contents are not uploaded.

## Additional Test Scenarios

- Valid manifest fixture.
- Malformed manifest fixture.
- Blocked path fixture.
- Recommendation influence test.
- Public/private boundary regression.

### PR DESCRIPTION
**Pull request by kiannidev**
Fixes #118

## Summary

- Adds a machine-readable **Gittensory repo focus manifest** at `.gittensory.yml` (wanted backend/MCP paths, blocked website/lovable areas, test expectations, public vs private notes).
- Extends focus-manifest parsing to support **YAML** (`.gittensory.yml` first, then JSON candidates) with safe malformed-manifest warnings.
- Bundles a self-repo fallback (`src/config/gittensory-repo-focus-manifest.ts`) so decision packs, preflight, and maintainer APIs work before the file is on `main`.
- Exposes maintainer-only `GET` / `PUT` `/v1/repos/{owner}/{repo}/focus-manifest` returning manifest + compiled policy.

## Scope

- [x] This PR is focused and does not mix unrelated backend, UI, MCP, docs, dependency, and deploy changes.
- [x] This follows `CONTRIBUTING.md` and does not reintroduce GitHub Pages, VitePress, `site/`, or `CNAME`.
- [x] I linked an issue, or this is small enough that the summary explains why an issue is not needed.

## Validation

- [x] `git diff --check`
- [x] `npm run actionlint`
- [x] `npm run typecheck`
- [x] `npm run test:coverage` locally; global coverage stays at or above **97%** for lines, statements, functions, and branches (aim for **98%+** branch coverage locally so CI variance does not fail near the threshold)
- [x] `npm run test:workers`
- [x] `npm run build:mcp`
- [x] `npm run test:mcp-pack`
- [x] `npm run ui:openapi:check`
- [x] `npm run ui:lint`
- [x] `npm run ui:typecheck`
- [x] `npm run ui:build`
- [x] `npm audit --audit-level=moderate`
- [x] New or changed behavior has unit/integration tests for new branches, fallback paths, and sanitizer boundaries

If any required check was skipped, explain why:

-

**New / updated tests:** `test/unit/gittensory-focus-manifest.test.ts`, `test/unit/focus-manifest.test.ts`, `test/unit/focus-manifest-loader.test.ts`.

## Safety

- [x] No secrets, wallet details, hotkeys, coldkeys, user PATs, private keys, raw trust scores, private rankings, or private maintainer evidence are exposed.
- [x] Public GitHub text stays sanitized, low-noise, and does not imply compensation guarantees or optimization tactics.
- [x] Auth, cookie, CORS, GitHub App, Cloudflare, or session changes include negative-path tests.
- [x] API/OpenAPI/MCP behavior is updated and tested where needed.
- [x] UI changes use live API data or real empty/error/loading states, not production mock/demo fallbacks.
- [x] Visible UI changes include screenshots or a short recording.
- [x] Public docs/changelogs are updated where needed; changelogs are only edited for release-prep PRs.

## Notes

- Parent roadmap: #82.
- Self-dogfood uses `GITTENSORY_DRIFT_ISSUE_REPO` when set; otherwise `JSONbored/gittensory`.
- `GITTENSORY_REPO_FOCUS_MANIFEST_YAML` must stay in sync with `.gittensory.yml` (enforced by unit test).

### CODE REVIEW

**Review by JSONbored — CHANGES_REQUESTED** [review]
@kiannidev the manifest/parser/API shape is useful, but I cannot merge this policy as written.

A few notes:
- The main blocker is `.gittensory.yml` treating `apps/gittensory-ui/` as a blocked path.
- That would classify legitimate focused control-panel work as blocked, which conflicts with how this repo currently accepts UI work tied to operator analytics, safety, and release workflows.
- The API/OpenAPI direction is fine, but the policy semantics need to match the repo.

Required changes:
- Rework the focus manifest so UI work is discouraged/conditional where appropriate, not categorically blocked.
- Update tests to cover allowed focused UI changes and genuinely blocked surfaces separately.
- Keep the manifest language aligned with CONTRIBUTING instead of creating a stricter hidden policy.

Validation expected:
- Rerun the focused manifest/parser/API tests and any OpenAPI check touched by this PR.

**Comment by ghost** [comment]
> [!NOTE]
> ## Gittensory Gate skipped
> PR closed before full evaluation. No late first comment was created.
>
> | Signal | Result | Evidence | Action |
> | --- | --- | --- | --- |
> | Gate result | ⚠️ Skipped | JSONbored/gittensory#389 is no longer open. | No action. |

---
Checked by [Gittensory](https://github.com/JSONbored/gittensory), a quiet PR intelligence layer for OSS maintainers.

**Comment by kiannidev** [comment]
@JSONbored Thanks for the review — reworked the manifest policy and merged latest `main`.

**Policy changes**
- Removed `apps/gittensory-ui/` from `blockedPaths`; added it to `wantedPaths` so focused control-panel UI work (operator analytics, safety, release workflows) is in scope per CONTRIBUTING.
- Kept genuinely blocked surfaces only: `site/`, `CNAME`, and `**/lovable/**`.
- Updated `publicNotes` / `maintainerNotes` to discourage cosmetic UI-only polish without API wiring, not all UI work.

**Tests**
- Focused UI paths (`apps/gittensory-ui/src/routes/app.operator.tsx`) → `manifest_preferred_path`.
- Retired surfaces (`site/`, `CNAME`, lovable) → `manifest_blocked_path`.
- Added `test/unit/routes-focus-manifest.test.ts` for GET/PUT route auth and persistence.

**Merge**
- Resolved conflicts with `main`; kept focus-manifest routes and onboarding-pack preview.
- `npm run ui:openapi:check` and `npm run test:coverage` pass locally (97% branch coverage).

Ready for another look.

**Review by JSONbored — CHANGES_REQUESTED** [review]
@kiannidev some merge conflicts to fix before I can accurately review this, please.

File: src/api/routes.ts [hunk 1]
```diff
@@ -1491,6 +1492,23 @@ export function createApp() {
...
+    if (forbidden) return forbidden;
+    const manifest = await loadRepoFocusManifest(c.env, fullName, { refresh: c.req.query("refresh") === "true" });
+    return c.json({ repoFullName: fullName, manifest, policy: compileFocusManifestPolicy(manifest) });
+  });
+
+  app.put("/v1/repos/:owner/:repo/focus-manifest", async (c) => {
+    const fullName = `${c.req.param("owner")}/${c.req.param("repo")}`;
+    const forbidden = await requireAppRole(c, ["maintainer", "owner", "operator"]);
+    if (forbidden) return forbidden;
+    const body = (await c.req.json().catch(() => null)) ?? {};
```
Comment (JSONbored) [inline src/api/routes.ts]: Malformed JSON is coerced to `{}` and then persisted as an `api_record`, so a bad maintainer PUT can erase/replace the focus manifest while returning 200. The new test at `test/unit/routes-focus-manifest.test.ts:77` locks in that behavior. This should reject invalid JSON with 400 instead of treating it as an empty manifest update.

**Review by JSONbored — CHANGES_REQUESTED** [review]
Requesting changes. The branch is still unmergeable against current `main`, and there is also a concrete API behavior issue in the focus-manifest PUT handler. No merge or close action taken.

**Review by JSONbored — CHANGES_REQUESTED** [review]
@kiannidev this is much closer: the manifest policy and malformed-JSON behavior are no longer the blocker.

A few notes:
- The remaining blocker is route authorization, not the manifest format.
- `/v1/repos/:owner/:repo/focus-manifest` GET/PUT uses `requireAppRole(...)` but does not apply `requireSessionRepoAccess(...)` for browser sessions.
- Comparable repo-scoped app routes check the authenticated session against the target repo before returning or mutating repo-specific maintainer data.
- As written, a maintainer/owner session with access to one repo can read or update another repo's focus manifest if they have the broad role.

Required changes:
- Add the same session repo-scope guard used by comparable repo routes.
- Cover allowed same-repo sessions, forbidden cross-repo sessions, static-token/operator access, and malformed JSON.
- Keep the current manifest policy semantics.

Validation expected:
- Rerun focused focus-manifest route tests, parser/loader tests, and OpenAPI checks.

**Comment by kiannidev** [comment]
@JSONbored Thanks — agreed, the missing repo-scope guard was the blocker.

**Route authorization**

- GET/PUT /v1/repos/:owner/:repo/focus-manifest now call requireSessionRepoAccess for browser sessions, matching onboarding-pack and other repo-scoped maintainer routes.
- Added isRepoFocusManifestPath() to canSessionAccessPath() so owner/maintainer sessions reach the handler; repo scope is enforced there instead of failing early with insufficient_role.
- Static-token callers are unchanged (no session → no repo-scope check).

**Tests (test/unit/routes-focus-manifest.test.ts)**

- Allowed same-repo owner session GET/PUT
- Forbidden cross-repo owner session → 403 forbidden_repo
- Operator session can access any repo
- Static-token access + malformed JSON PUT → 400 invalid_json

**Validation**

- npm run typecheck
- npm run test:coverage (97.02% branches)
- npm run ui:openapi:check
- Focus-manifest route, parser, and loader tests

Manifest policy semantics unchanged. Ready for another look.

**Review by JSONbored — CHANGES_REQUESTED** [review]
@kiannidev the source changes are much closer now, and the repo-scoped session tests are the right fix for the earlier auth concern.

A few notes:
- I tried GitHub's update-branch rebase before reviewing, and GitHub reported a rebase conflict between the base and head.
- I am not going to resolve contributor branch conflicts from the maintainer side.
- Once rebased cleanly, this still looks like the right shape for the focus-manifest route.

Required changes:
~~~text
- Rebase this branch onto current main yourself.
- Preserve the focus-manifest GET/PUT session access tests, including same-repo owner, cross-repo owner, and operator coverage.
- Refresh generated OpenAPI/UI artifacts after the rebase.
- Re-run the focused focus-manifest tests and full validation.
~~~

Validation expected:
~~~text
- git diff --check
- focused routes-focus-manifest/focus-manifest tests
- ui:openapi:check
- full validate
~~~

**Comment by kiannidev** [comment]
@JSONbored Thanks — rebased onto current main and resolved the conflicts on my side.

**Rebase**

- Merged cleanly with current main, keeping focus-manifest routes alongside self-dogfood registration-pack and contributor-issue-draft endpoints from main.
- Regenerated apps/gittensory-ui/public/openapi.json after the rebase.

**Session access tests preserved**

- Same-repo owner GET/PUT
- Cross-repo owner → 403 forbidden_repo
- Operator session access to any repo
- Static-token access + malformed JSON PUT → 400 invalid_json

**Validation**

- git diff --check
- Focus-manifest route / parser / loader tests
- npm run ui:openapi:check
- npm run validate (97.05% branch coverage)

Manifest policy semantics unchanged. Ready for another look.

**Comment by kiannidev** [comment]
HI, @JSONbored 
Please review this PR..

Thanks

**Review by JSONbored — CHANGES_REQUESTED** [review]
@kiannidev this is now in good shape.

A few notes:
- The manifest policy no longer blocks legitimate focused UI work.
- GET/PUT focus-manifest routes now apply repo-scoped session access.
- The malformed JSON and OpenAPI/generated artifact concerns are addressed.
- Local merge-tree against current `main` is clean.
- No code changes requested.

Before I can merge this, please fix failing CI.

**Comment by kiannidev** [comment]
HI, @JSONbored

I fixed the CI error and resolved conflicts..
Please review this PR..

Thanks

**Review by JSONbored — APPROVED** [review]
@kiannidev this is good to land.

A few notes:
- The focus manifest now has the repo-scoped session guard on GET and PUT.
- The self-repo fallback, YAML parsing, OpenAPI registration, and policy compilation are covered by focused tests.
- The manifest keeps maintainer notes out of public guidance and keeps retired surfaces blocked.
- Superagent is neutral, not a blocker here, and the non-Gittensory checks are green.

No code changes requested.
